api/src/resolver_repo.ts

```diff
@@ -515,10 +515,50 @@ async function copyRepo(_, { repoId }, { userId }) {
   return id;
 }
 
+/**
+ * create yDoc snapshot upon request.
+ */
+async function addRepoSnapshot(_, { repoId, message }) {
+  const repo = await prisma.repo.findFirst({
+    where: { id: repoId },
+    include: {
+      owner: true,
+      collaborators: true,
+    },
+  });
+  if (!repo) throw Error("Repo not exists.");
+  if (!repo.yDocBlob) throw Error(`yDocBlob on ${repoId} not found`);
+  const snapshot = await prisma.yDocSnapshot.create({
+    data: {
+      id: await nanoid(),
+      yDocBlob: repo.yDocBlob,
+      message: message,
+      repoId: repoId,
```

Comment:
Use the [connect](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-a-single-record) clause to connect the two tables.